tensorflow/compiler/xla/service/gpu/ir_emitter_nested.cc

```diff
@@ -625,14 +644,24 @@ Status EmitAtomicOperationUsingCAS(llvm::IRBuilder<>* builder,
         builder->CreatePtrToInt(output_address, address_int_type);
     llvm::Value* mask = llvm::ConstantInt::get(address_int_type, 3);
     llvm::Value* offset = builder->CreateAnd(atomic_memory_address, mask);
-    mask = llvm::ConstantInt::get(address_int_type, -4);
-    atomic_memory_address = builder->CreateAnd(atomic_memory_address, mask);
-    atomic_memory_address =
-        builder->CreateIntToPtr(atomic_memory_address, atomic_address_type);
-    binop_output_address = builder->CreateAdd(
-        builder->CreatePtrToInt(cas_new_output_address, address_int_type),
-        offset);
-    binop_output_address = builder->CreateIntToPtr(
+
+    // inttoptr (and (ptrtoint %output_address) -4)
```

Comment:
This is for CAS? Why do we need to add code changes for CAS?

Reply:
Comment below.